### PR TITLE
KAFKA-12697: Add FencedBrokerCount and ActiveBrokerCount metrics to the QuorumController

### PR DESCRIPTION
**Pull request by ryandielhenn**
Adding FencedBrokerCount and ActiveBrokerCount metrics to the QuorumController

Tested using MockControllerMetrics.

https://issues.apache.org/jira/browse/KAFKA-12882

https://cwiki.apache.org/confluence/display/KAFKA/KIP-748%3A+Add+Broker+Count+Metrics#KIP748:AddBrokerCountMetrics

### CODE REVIEW

File: metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java [hunk 1]
```diff
@@ -249,6 +256,10 @@ public void replay(RegisterBrokerRecord record) {
             features.put(feature.name(), new VersionRange(
                 feature.minSupportedVersion(), feature.maxSupportedVersion()));
         }
+       
+        if (record.fenced()) {
+            controllerMetrics.setFencedBrokerCount(controllerMetrics.fencedBrokerCount() + 1);
```
Comment (cmccabe) [inline metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java]: We need to update this based on what the previous registration was, if any. If the previous registration was also fenced, we do not want to increase the fenced broker count.

It is also possible for the new registration to start as unfenced. Please look at the record definitions.

It would be useful to have a helper function that took as an argument the previous registration (or null) and the new registration (or null), and updated the metrics accordingly.

File: metadata/src/main/java/org/apache/kafka/controller/QuorumControllerMetrics.java [hunk 2]
```diff
@@ -26,28 +26,36 @@
...
-    private final static MetricName GLOBAL_TOPIC_COUNT = getMetricName(
-        "KafkaController", "GlobalTopicCount");
-    private final static MetricName GLOBAL_PARTITION_COUNT = getMetricName(
-        "KafkaController", "GlobalPartitionCount");
-    private final static MetricName OFFLINE_PARTITION_COUNT = getMetricName(
-        "KafkaController", "OfflinePartitionsCount");
-    private final static MetricName PREFERRED_REPLICA_IMBALANCE_COUNT = getMetricName(
-        "KafkaController", "PreferredReplicaImbalanceCount");
-
+    private final static MetricName ACTIVE_CONTROLLER_COUNT = new MetricName(
```
Comment (cmccabe) [inline metadata/src/main/java/org/apache/kafka/controller/QuorumControllerMetrics.java]: why are we changing from `getMetricName` to `new MetricName`? Should this be a separate PR?

**Comment by cmccabe** [comment]
thanks for this, @dielhennr. I left some comments.

File: metadata/src/main/java/org/apache/kafka/controller/QuorumControllerMetrics.java (hunk 2 above)
Comment (ryandielhenn) [reply metadata/src/main/java/org/apache/kafka/controller/QuorumControllerMetrics.java]: This was a mistake when I merged trunk into my branch. I reverted it.

**Review by cmccabe — APPROVED** [review]
LGTM.

Thanks, @dielhennr
